ee/hogai/PROMPTING_GUIDE.md

```diff
@@ -0,0 +1,237 @@
+# LLM prompting guide for PostHog
+
+You know code, now meet prompts. This guide explains how we make effective LLM prompts in PostHog's AI features.
+
+## Standard building blocks
+
+Always use `MaxChatOpenAI` instead of plain LangChain `ChatOpenAI`:
+
+```python
+from ee.hogai.llm import MaxChatOpenAI
+
+# ✅ Correct - auto-injects user/project/org context
+llm = MaxChatOpenAI(user=user, team=team, model="gpt-4.1")
+
+# ❌ Wrong - missing PostHog context
+llm = ChatOpenAI(model="gpt-4.1") 
+```
+
+`MaxChatOpenAI` automatically injects context into every prompt:
+- Project name and timezone
+- Organization name  
+- User name and email
+- Current project datetime
+
+This context appears at the end of system messages.
+
+## The anatomy of a PostHog prompt
+
+PostHog prompts typically follow a structure like this:
+
+```python
+SYSTEM_PROMPT = """
+<agent_info>
+You are Max, PostHog's AI assistant...
+Your role and personality description.
+</agent_info>
+
+<instructions>
+Specific task instructions and guidelines.
+</instructions>
+
+<constraints>
+What the agent should and shouldn't do.
+</constraints>
+
+<examples>
+Few-shot examples demonstrating the expected behavior.
+</examples>
+
+{{{dynamic_context}}}
+""".strip()
+```
+
+As you see, we use non-nested XML tags to clearly delineate sections.
+
+### Variable templating with Mustache
+
+PostHog uses Mustache templating for dynamic content:
```

Comment:
FIY when using `o3` and setting instructions, you can't use Mustache as you're just passing the raw formatted string, it doesn't go through LangGraph's fancy formatting.

Reply:
Hmm, will skip this complexity for now as current `master` doesn't have an example, and we also the term `instructions` in our own meaning in a few prompt templates